00-Login-Kt/app/src/main/java/com/auth0/sample/MainActivity.kt

```diff
@@ -96,11 +136,71 @@ class MainActivity : AppCompatActivity() {
                     }
 
                     override fun onSuccess(profile: UserProfile?) {
-                        binding.userProfile.setText(
-                                "Name: ${profile?.name}\n" +
-                                "Email: ${profile?.email}")
+                        cachedUserProfile = profile;
+                        updateUI()
                     }
         }) }
     }
 
+    private fun getUserMetadata() {
+        // Get the access token so that we can make calls to the management API
+        cachedCredentials?.accessToken?.let { accessToken ->
+            // Get the user ID and call the full getUser Management API endpoint, to retrieve the full profile information
+            cachedUserProfile?.getId()?.let { userId ->
```

Comment:
if you are sure this `getUserMetadata` method is only called when those 2 nullable vars are set (remember you enable/disable the buttons as well when the state changes), I'd remove the `let`s as they just _indentate_ the code to the right for nothing. Or at least this second one for the ID. The ID that comes from userinfo is the sub claim, always present because of `openid` scope.

Reply:
I can do that. The one thing `let` does do is allow me to access `accessToken` as a `String` instead of a `String?`.

With your change, the resulting line ends up like this:

```kotlin
val usersClient = UsersAPIClient(account, cachedCredentials!!.accessToken ?: "")
```

Is that the way to write this?

Reply:
Sorry, just realised I can do this:

```kotlin
val usersClient = UsersAPIClient(account, cachedCredentials!!.accessToken!!)
```

What would you prefer?

Reply:
if the method is never going to be invoked when the credentials are null, I'd keep this last snippet you posted. I think it's cleaner. You could add a comment explaining that `cachedCredentials` is never null when this method gets called.

```
val usersClient = UsersAPIClient(account, cachedCredentials!!.accessToken!!)
```